src/symbol/dwarf_utils.rs

```diff
@@ -1,7 +1,94 @@
-use gimli::{DebuggingInformationEntry, Dwarf, Unit};
+use gimli::{DebuggingInformationEntry, Dwarf, Unit, UnitOffset, ValueType};
 
 use super::Reader;
 
+macro_rules! dwarf_attr {
+    (str($dwarf:ident,$unit:ident) $entry:ident.$name:ident || $missing:ident) => {
+        if let Some(attr) = $entry.attr(gimli::$name)? {
+            dwarf_attr_not_missing_action!($missing, $dwarf.attr_string(&$unit, attr.value())?)
```

Comment:
Nitpick - I'd suggest using a more descriptive name, e.g.
```suggestion
            dwarf_attr_exists_action!($missing, $dwarf.attr_string(&$unit, attr.value())?)
```